NSObject_Rx/DisposeBagable.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  DisposeBagable.swift
+//  NSObject-Rx
+//
+//  Created by Thibault Wittemberg on 2017-08-25.
+//  Copyright © 2017 RxSwiftCommunity. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import ObjectiveC
+
+fileprivate struct AssociatedKeys {
+    static var rxDisposeBag = "rxDisposeBag"
+}
+
+/// each DisposeBagable offers a unique Rx DisposeBag instance
+public protocol DisposeBagable {
+
+    /// a unique Rx DisposeBag instance
+    var rxDisposeBag: DisposeBag { get set }
+}
+
+extension DisposeBagable {
+
+    private func doLocked(forClosure closure: () -> Void) {
+        objc_sync_enter(self); defer { objc_sync_exit(self) }
+        closure()
+    }
+
+    public var rxDisposeBag: DisposeBag {
+        get {
+            var disposeBag: DisposeBag!
+            doLocked {
+                let lookup = objc_getAssociatedObject(self, &AssociatedKeys.rxDisposeBag) as? DisposeBag
```

Comment:
Interesting – I didn't expect associated objects to work with non-NSObject subclasses. You've verified this works?

Reply:
I will add a Unit test for this

Reply:
Good idea 👍

Reply:
Done, there is a new Test for that. It works.